test/checks/lists/only-listitems.js

```diff
@@ -18,12 +18,30 @@ describe('only-listitems', function () {
 		assert.isFalse(checks['only-listitems'].evaluate.apply(checkContext, checkArgs));
 	});
 
+	it('should return false if the list has only spaces as content', function () {
+		var checkArgs = checkSetup('<ol id="target">   </ol>');
+
+		assert.isFalse(checks['only-listitems'].evaluate.apply(checkContext, checkArgs));
+	});
+
 	it('should return false if the list has whitespace', function () {
 		var checkArgs = checkSetup('<ol id="target"><li>Item</li>    </ol>');
 
 		assert.isFalse(checks['only-listitems'].evaluate.apply(checkContext, checkArgs));
 	});
 
+	it('should return false if the list has only an element with role listitem', function () {
+		var checkArgs = checkSetup('<ol id="target"><div role="listitem">A list</div></ol>');
```

Comment:
The problem with this one is it is an HTML validation problem because `<ol>` is only supposed to have `<li>` and `script supporting` children https://www.w3.org/TR/html52/grouping-content.html#the-ol-element

Reply:
@dylanb I think that's on me - I replied to it saying that Axe shouldn't be a validator. As long as it works in AT, it shouldn't matter if it's valid HTML or not.